### ISSUE TITLE
Probably a bug in gpu.cpp

### ISSUE THREAD

**Issue by Keksov** [fireice-uk/xmr-stak#1155]
To my mind, the declaration `size_t g_thd` (just a local variable) inside `if(ctx->compMode)` prevents outer `g_thd` from proper rounding. 
```
	size_t g_thd = g_intensity;

	if(ctx->compMode)
	{
		// round up to next multiple of w_size
		size_t g_thd = ((g_intensity + w_size - 1u) / w_size) * w_size;
		// number of global threads must be a multiple of the work group size (w_size)
		assert(g_thd%w_size == 0);
	}
```
Should be just `g_thd = ((g_intensity + w_size - 1u) / w_size) * w_size;`

**Comment by psychocrypt** [comment]
thx for pointing to it. You are right the new declaration within the local
scope is wrong. I will fix it soon.

**Comment by psychocrypt** [comment]
fixed with #1156
